app/models/repository.rb

```diff
@@ -29,12 +29,17 @@
 
 class Repository < ActiveRecord::Base
   include Redmine::Ciphering
+  include OpenProject::Scm::ManageableRepository
 
   belongs_to :project
   has_many :changesets, order: "#{Changeset.table_name}.committed_on DESC, #{Changeset.table_name}.id DESC"
 
   before_save :sanitize_urls
 
+  # Managed repository lifetime
+  after_save :create_managed_repository, if: Proc.new { |repo| repo.managed? }
+  after_destroy :delete_managed_repository, if: Proc.new { |repo| repo.managed? }
```

Comment:
> (It only creates or removes one directory anyway).

I think this depends on the SCM implementation. If it requires to call a web service to have some repo created on a different machine, using delayed job makes a lot of sense. However, it is true for our local setup. But if we decide to abandon delayed job for our setup, it should still be possible for others to use it.

I might not have understood your proposal regarding the service class and it's transaction:
> We could then let a service class create a transaction and provide some kind of migration-style functionality (up, down) that will service both for the regular creating / removal flow, as well as rolling back from an error in the transaction.

What will be this service class's responsibility?
* Creating the repo in DB?
* Creating the repo on FS?